matplotcheck/autograde.py

```diff
@@ -0,0 +1,89 @@
+# Wrapper functions for matplotcheck functions that run
+# and track the passing status of test functions and
+# provide formatted results of tests
+
+
+def run_test(
+    func,
+    points,
+    *args,
+    correct_message="default correct",
+    error_message="default error",
+    **kwargs
+):
+    """Runs a pre-defined test function and creates a dictionary
+    containing the results of the test
+
+    Parameters
+    ---------
+    func: name of pre-defined test function to run
+    points: number of points assigned for passing test
+    args: arguments provided to test function
+    correct_message: string of custom message returned with passing test
+    error_message: string of custom message returned with failing test
+    kwargs: keyword arguments provided to test function
+
+    Returns
+    -------
+    results: dictionary with the following key:value pairs
```

Comment:
i haven't seen how people handle a key-value pair in the docs before so maybe see if there is a standard for this. it looks a bit like parameter documentation here but because i am not sure what hte standard is i can't comment on it.

Reply:
@lwasser OK after checking the [numpy docs](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard) and [PEP 257 on docstring conventions](https://www.python.org/dev/peps/pep-0257/), I cannot find a standard for listing keys-values in dictionaries. I've posted on our python-for-dayzzzz slack channel. We could post on stackoverflow, but it seems that others have the same question:
- [related question about long docstring](https://stackoverflow.com/questions/25122378/docstring-style-should-i-really-have-it-this-long)
- [related question about specifying keys in docstrings](https://stackoverflow.com/questions/55059571/add-dictionary-keys-in-docstring-sphinx)
- [related question about docstrings with dicts parameter](https://stackoverflow.com/questions/26712533/formatting-python-docstrings-for-dicts)

Based on all this and our standard, I've made a suggestion for specifying the keys. Let me know what you think. 

Last, when I run `type()` on the keys in `results`, the type for all of them (including points or pass) are actually `str`. Should we update these types to the expected types that I listed in the docstring, or update the docstring to indicate that the output keys are all type `str`?